### PR TITLE
Add a documentdb storage backend.

### PR DESCRIPTION
**Pull request by dlorenc**
This allows for flexible storage and querying of fully-nested types. Full lookups
on each property in the ITE-6 provenance payload, for example.

### CODE REVIEW

**Comment by dlorenc** [comment]
You can query all the properties of the TaskRun or ITE-6 format with this:
image

File: pkg/chains/storage/docdb/docdb.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+// StorePayload implements the Payloader interface.
+func (b *Backend) StorePayload(signed []byte, signature string, key string) error {
+	// We need two object names: the signature and the payload. We want to make these unique to the UID, but easy to find based on the
+	// name/namespace as well.
+	// $bucket/taskrun-$namespace-$name-$uid/$key.signature
+	// $bucket/taskrun-$namespace-$name-$uid/$key.payload
+	root := fmt.Sprintf("taskrun-%s-%s-%s", b.tr.Namespace, b.tr.Name, b.tr.UID)
+	sigName := path.Join(root, fmt.Sprintf("%s.signature", key))
```
Comment (kommendorkapten) [inline pkg/chains/storage/docdb/docdb.go:75]: `sigName` is never used except for the log message. In the signed document, `Name` is set to `key` (which is the last part of `sigName`), that's not where `sigName` was meant to go?

Comment (kommendorkapten) [reply pkg/chains/storage/docdb/docdb.go:75]: Also two names are mentioned in the comment, but only one (`sigName`) is generated.

Comment (dlorenc) [reply pkg/chains/storage/docdb/docdb.go]: Thanks! copy/paste mistake from the GCS storage one :)

**Review by priyawadhwa — APPROVED** [review]
LGTM! The annotations in the README [here](https://github.com/tektoncd/chains/blob/main/README.md#usage) just need to be updated to include the UID

**Review by font — APPROVED** [review]
Nice!

/lgtm

**Pull request merged**